internal/test/framework/windowsvm.go

```diff
@@ -51,57 +52,117 @@ type TestWindowsVM interface {
 	types.WindowsVM
 }
 
-// newWindowsVM creates and sets up a Windows VM in the cloud and returns the WindowsVM interface that can be used to
-// interact with the VM. If credentials are passed then it is assumed that VM already exists in the cloud and those
-// credentials will be used to interact with the VM. If no error is returned then it is guaranteed that the VM was
-// created and can be interacted with. If skipSetup is true, then configuration steps are skipped.
-func newWindowsVM(imageID, instanceType string, credentials *types.Credentials, skipSetup bool) (TestWindowsVM, error) {
-	w := &testWindowsVM{}
-	var err error
-
-	cloudProvider, err = cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
-		imageID, instanceType, sshKey, privateKeyPath)
+//createMachineSet() gets the generated MachineSet configuration from cloudprovider package and creates a MachineSet
+func (f *TestFramework) createMachineSet() error {
+	cloudProvider, err := providers.NewCloudProvider(sshKey)
+	if err != nil {
+		return fmt.Errorf("error instantiating cloud provider %v", err)
+	}
+	machineSet, err := cloudProvider.GenerateMachineSet(true, 1)
+	if err != nil {
+		return fmt.Errorf("error creating Windows MachineSet: %v", err)
+	}
+	f.machineSet = machineSet
+	log.Print("Creating Machine Sets")
+	err = f.client.Create(context.TODO(), machineSet)
 	if err != nil {
-		return nil, fmt.Errorf("error instantiating cloud provider %v", err)
+		return fmt.Errorf("unable to create MachineSet %v", err)
 	}
+	log.Print("Created Machine Sets")
+	return nil
+}
 
-	if credentials == nil {
-		windowsVM, err := cloudProvider.CreateWindowsVM()
+// waitForMachines() waits until all the machines required are in Provisioned state. It returns an array of all
+// the machines created. All the machines are created concurrently.
+func (f *TestFramework) waitForMachines(vmCount int) ([]mapi.Machine, error) {
+	log.Print("Waiting for Machine Sets ")
+	windowsOSLabel := "machine.openshift.io/os-id"
+	var provisionedMachines []mapi.Machine
+	// it takes approximately 12 minutes in the CI for all the machines to appear.
+	timeOut := 12 * time.Minute
```

Comment:
Shouldn't the VM count be factored into this?

Reply:
need to check that. As all VMs are created simultaneously..

Reply:
Did you get a chance to check on this?

Reply:
yes, actually 12 min should be enough for all VMS to come up. 
They actually come up in parallel.